### ISSUE TITLE
Content Security Policy parameters not reset when app cache is cleared

### ISSUE THREAD

**Issue by degrashopper** [home-assistant/android#3420]
**Home Assistant Android app version(s):**
2023.1.1 Full
**Android version(s):**
android 12
**Device model(s):**
Samsung galaxy s10
**Home Assistant version:**
2023.3.2
**Last working Home Assistant release (if known):**
none
**Description of problem, include YAML if issue is related to notifications:**
When you clear the cache of the HA companion app after having to make a change to the content security policy parameters of the proxy where HA is running behind. The content security policy parameters are not reset. Only after clearing the memory of the app these parameters are reset. So every time you need to make a change to the content security policy you have to reconfigure the companion app. In fore example the chrome browser these content security policy parameters are reset when you clear the cache.

**Companion App Logs:**

```

```

**Screenshot or video of problem:**

**Additional information:**

**Comment by dshokouhi** [comment]
Dont think this is something that the app can solve, everything cached is cached by default and we have been asked to respect the caching policy set by the HA frontend. This sounds more like a Android OS issue than an app issue.  Have you tried looking at the logs that you skipped over to see if the error is something you can solve on the network end?

**Comment by degrashopper** [comment]
The content security policy is set in the proxy that protects HA from the outside. The app uses this policy. But if you change the policy on the server side and clear the cache of the client app it still won't use the new content security policy. The only way to do this is to clear the app memory and reconfigure the app. Only then the app will use the content security policy correctly. If you do this via a browser then you only have to clear the cache to pick up the new CSP. So there is no error to look for.

**Comment by dshokouhi** [comment]
The android app respects the caching level set by the HA frontend which is actually very heavy caching so that is working as intended. Clearing app cache is not the same as we use WebView which has different caching and I do not think that cache is the same as you are expecting here. Have you tried to force stop the app and see if that forces it to refresh as you would expect?

**Comment by degrashopper** [comment]
@dshokouhi after your question I tested a forced app stop. This won't refresh/reset the CSP either. CSP is not part of the HA frontend but a HTTP response header and is used to enhance the security of the webpage. I don't know how hard it is to use it in cache instead of memory. I'm just saying the CSP is not properly updated if you change it. And with the constant changes the HA team is doing on the frontend, you have to update the CSP settings otherwise the frontend is not working properly.

**Comment by dshokouhi** [comment]
> I don't know how hard it is to use it in cache instead of memory.

From the apps standpoint we do not control that, we just load the URL and respect the heavy caching rules set by the HA frontend as we were requested to do so.

This may be behaving as expected for a webview applications. I am going to close this request as this is not something officially documented or supported by HA.

**Comment by degrashopper** [comment]
Hmm okay. Thanks for your time.
